src/Statement/Grouper/StatementGrouper.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+interface StatementGrouper {
+
+	/**
+	 * @param StatementList $statements
+	 *
+	 * @return StatementList[] An associative array, mapping statement group identifiers to
```

Comment:
> where the values come from.

This is an interface. It can not know where this comes from. May be from some constructor parameter, may be some generated keys, depending on the implementation.